crd-generator/api/src/test/java/io/fabric8/crd/generator/CRDGeneratorTest.java

```diff
@@ -545,6 +546,38 @@ void checkGenerationMultipleVersionsOfCRDsIsDeterministic() throws Exception {
     assertTrue(outputDir.delete());
   }
 
+  @Test
+  void checkK8sValidationRules() throws Exception {
+    // generated CRD
+    final File outputDir = Files.createTempDirectory("crd-").toFile();
```

Comment:
If I understand this correctly, moving this test to `crd-generator/test` should remove the need for the "infrastructure" of creating the temp dir etc., can you verify?

Reply:
Yes, it's the same test strategy we already used a few times. For example:

- https://github.com/fabric8io/kubernetes-client/blob/main/crd-generator/api/src/test/java/io/fabric8/crd/generator/CRDGeneratorTest.java#L483
- https://github.com/fabric8io/kubernetes-client/blob/main/crd-generator/api/src/test/java/io/fabric8/crd/generator/CRDGeneratorTest.java#L513

Those kind of tests generate real YAML manifests in the temp directory and compare the resulting files with expected manifests from the test resources.

An already implemented utility assertion method is used to for the comparison, which ignores comments and empty lines:

https://github.com/fabric8io/kubernetes-client/blob/main/crd-generator/api/src/test/java/io/fabric8/crd/generator/CRDGeneratorTest.java#L548

The temp directory will be cleaned up if the test is successfully. If not, the temp directory won't be deleted which makes it easier to debug.

In my opinion those tests are more end-to-end and allow quickly adding edge cases. 
In the future those tests should be refactored to make it more reusable.

Reply:
it's fine, I didn't recall this pattern was already used here, disregard my comment 👍